builtin/game/item.lua

```diff
@@ -205,6 +207,16 @@ function core.get_node_drops(nodename, toolname)
 	return got_items
 end
 
+local non_player_name = "!nonplayer"
+local function user_name(user)
+	return user and user:get_player_name() or non_player_name
+end
+
+local function is_protected(pos, name, user)
+	return core.is_protected(pos, name) and
+		not (name ~= non_player_name and minetest.check_player_privs(user, "protection_bypass"))
```

Comment:
-> `core.check_player_privs`
This function also accepts a player name as argument.